src/Microsoft.AspNetCore.Mvc.Core/Internal/FileStreamResultExecutor.cs

```diff
@@ -18,18 +23,72 @@ public FileStreamResultExecutor(ILoggerFactory loggerFactory)
 
         public Task ExecuteAsync(ActionContext context, FileStreamResult result)
         {
-            SetHeadersAndLog(context, result);
-            return WriteFileAsync(context, result);
+            var fileLength = 0L;
```

Comment:
nullable? it won't be set for !CanSeek. It looks like this code sets Content-Length to 0 for non-seekable streams.